File: web/src/game/actions/Arrival.ts
import { tsm, uint } from '../../types';
import { Action } from './Action';
import { Creature } from '../Creature';
import { Npc } from '../Npc';
import { Metrics } from '../Metrics';

export class Arrival implements Action {
  readonly id: uint;
  readonly creature: Creature;
  readonly time: tsm;

  constructor(data: any) {


    const m     = new Metrics(data.creature.metrics.life, data.creature.metrics.name);
    this.creature = new Npc(m, data.creature.x, data.creature.y);


  }


  // constructor(source: Creature) {
  //   this.id     = nextId();
  //   this.time   = Date.now();//fixme take it from performance*
  //   this.source = source;
  // }
}

File: server/src/main/kotlin/ballad/server/game/Arrival.kt
package ballad.server.game

import ballad.server.Tsm

data class Arrival(
    override val x: Int,
    override val y: Int,
    override val time: Tsm,
    val creature: Creature
) : Action {
    constructor(time: Tsm, c: Creature) : this(c.x, c.y, time, c)
}